### PR TITLE
Imviz: Load NDData and ndarray

### PR DESCRIPTION
**Pull request by pllim**
The less controversial part of #600 (JDAT-1486). This adds support for `astropy.nddata.NDData` and `numpy.ndarray` in Imviz.

With this patch, the following is now possible (I copy-pasted the code from notebook cells):

```python
import warnings
warnings.simplefilter('ignore')
```

```python
from jdaviz import Imviz

imviz = Imviz()
viewer = imviz.app.get_viewer('viewer-1')

imviz.app
```

```python
import numpy as np
from astropy.nddata import NDData

a = np.zeros((10, 10))
a[1:-1, 1] = 1
a[1:-1, 8] = 1
a[1, 1:-1] = 1
a[8, 1:-1] = 1

imviz.load_data(a, data_label='ndarray')

b = NDData(a, mask=~a.astype(np.bool_))
imviz.load_data(b, data_label='nddata')
```

Screenshot 2021-05-14 101032

Screenshot 2021-05-14 101105

### CODE REVIEW

**Review by astrofrog — APPROVED** [review]
This seems sensible from a technical point of view. I'll leave it to a PO to review whether the behavior makes sense.

**Review by eteq — CHANGES_REQUESTED** [review]
This is looking great to me!  But inspired by the jpg/png support, can you also implement support for RAW files from my favorite android phone and maybe my friends's DSLR camera?

Just Kidding! :wink: 

More seriously, one other thing is missing: a docstring for `load_data` that explains all the supported formats. In particular, I think it's good to point out jpg/png will be loaded but will be flattened down to greyscale. That's a fine choice to make, I think it's just important it be written down in the docstrings.

File: jdaviz/configs/imviz/helper.py
Comment (pllim) [inline jdaviz/configs/imviz/helper.py:19]: @eteq , clearer now?

**Comment by pllim** [comment]
> jpg/png will be loaded but will be flattened down to greyscale

Yes, this is intentional. If you want RGB[A] layers preserved, you can ask Cubeviz to support that format. 😉 

> RAW files from my favorite android phone and maybe my friends's DSLR camera?

If there is a Python package that can do any of this and you don't mind it being declared as optional dependency in Jdaviz, then I don't mind adding the support, though it is not MVP. 😉 

Maybe won't be so bad to also add a YouTube plugin to stream Baby Shark... I am sure @astrofrog knows how to do it!

**Comment by rosteen** [comment]
@eteq would you confirm that your requested changes have been addressed, please?

File: jdaviz/configs/imviz/helper.py
Comment (eteq) [inline jdaviz/configs/imviz/helper.py:45]:
```suggestion

        Notes
        -----
        When loading image formats that support RGB color like jpg or png, the
        files are  converted to greyscale. This is done following the algorithm
        of `skimage.color.rgb2grey`, which involves weighting the channels as
        ``0.2125 R + 0.7154 G + 0.0721 B``. If you prefer a different weighting,
        you can use `skimage.io.imread` and from that produce your own greyscale
        image.
```

File: jdaviz/configs/imviz/helper.py
Comment (eteq) [reply jdaviz/configs/imviz/helper.py:19]: yes definitely!  Thanks.

**Review by eteq — APPROVED** [review]
LGTM now,  aside from one suggestion to clarify what's going on with the greyscale!

File: jdaviz/configs/imviz/helper.py
Comment (pllim) [reply jdaviz/configs/imviz/helper.py:45]: I didn't know grayscale weighting is so controversial. Thanks for the clarification! 😸

**Comment by pllim** [comment]
Given the approvals, I will merge when CI passes. Thanks!

**Pull request merged**
